dotnet/src/Connectors/Connectors.Memory.Weaviate/Http/ApiSchema/BatchDeleteRequest.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Net.Http;
+
+namespace Microsoft.SemanticKernel.Connectors.Weaviate;
+
+internal sealed class BatchDeleteRequest
+{
+    private readonly string _class;
+    private readonly string[] _ids;
+
+    private BatchDeleteRequest(string @class, string[] ids)
+    {
+        this._class = @class;
+        this._ids = ids;
+    }
+
+    public static BatchDeleteRequest Create(string @class, string[] ids)
+    {
+        for (int i = 0; i < ids.Length; i++)
+        {
+            ids[i] = ToWeaviateFriendlyId(ids[i]);
+        }
```

Comment:
It's better to avoid changing the input params into a method, since callers usually do not expect that unless that's the purpose of a method.
Consider instead doing `var friendlyIds = ids.Select(ToWeaviateFriendlyId).ToArray()`